main/webapp/modules/core/scripts/project/browsing-engine.js

```diff
@@ -198,6 +198,9 @@ BrowsingEngine.prototype.addFacet = function(type, config, options) {
          if (facet.facet.checkInitialHeight) {
            facet.facet.checkInitialHeight();
          }
+         if (facet.facet instanceof RangeFacet) {
+           facet.facet.render();
```

Comment:
In the future I would like to make facets properly extensible, so that extensions could define their own facets. With that in mind, I would generally try to avoid having code in the core of the code that are specific to a given facet. Would it make sense to render the facet regardless of its type, or do you see any complications with that?
I can imagine that facets defined by extensions could also want to be rendered when the facet tab is activated.

Reply:
I had similar thoughts. One issue is `render()` is not in the base class `Facet()`, and `ListFacet()` does not define the `render()` method.
 
 But we can still make it more general and less brittle by checking if the facet defines `render()`, and if so, call it.
 Otherwise, we still have a special case for `ListFacet`.

Reply:
Or have a default `render()` method in the base `Facet()` class?